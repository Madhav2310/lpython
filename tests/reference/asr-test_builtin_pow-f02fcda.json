{
    "basename": "asr-test_builtin_pow-f02fcda",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_builtin_pow.py",
    "infile_hash": "5696eb31b14c38dcd0c4d04d12af41c66c8efd457a54b9adbce00229",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_builtin_pow-f02fcda.stdout",
    "stdout_hash": "ecea0cba733dd9ccf0dbc689548686ab52feee5f91e8767d38042e5e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}