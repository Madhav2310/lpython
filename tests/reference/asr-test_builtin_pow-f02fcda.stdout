(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 97 {}) _lpython_main_program [] [(SubroutineCall 1 test_pow () [] ())] Source Public Implementation () .false. .false.), abs@__lpython_overloaded_0__abs: (ExternalSymbol 1 abs@__lpython_overloaded_0__abs 4 __lpython_overloaded_0__abs lpython_builtin [] __lpython_overloaded_0__abs Public), complex@__lpython_overloaded_9__complex: (ExternalSymbol 1 complex@__lpython_overloaded_9__complex 4 __lpython_overloaded_9__complex lpython_builtin [] __lpython_overloaded_9__complex Public), lpython_builtin: (IntrinsicModule lpython_builtin), main_program: (Program (SymbolTable 96 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), pow@__lpython_overloaded_0__pow: (ExternalSymbol 1 pow@__lpython_overloaded_0__pow 4 __lpython_overloaded_0__pow lpython_builtin [] __lpython_overloaded_0__pow Public), pow@__lpython_overloaded_1__pow: (ExternalSymbol 1 pow@__lpython_overloaded_1__pow 4 __lpython_overloaded_1__pow lpython_builtin [] __lpython_overloaded_1__pow Public), pow@__lpython_overloaded_2__pow: (ExternalSymbol 1 pow@__lpython_overloaded_2__pow 4 __lpython_overloaded_2__pow lpython_builtin [] __lpython_overloaded_2__pow Public), pow@__lpython_overloaded_3__pow: (ExternalSymbol 1 pow@__lpython_overloaded_3__pow 4 __lpython_overloaded_3__pow lpython_builtin [] __lpython_overloaded_3__pow Public), pow@__lpython_overloaded_4__pow: (ExternalSymbol 1 pow@__lpython_overloaded_4__pow 4 __lpython_overloaded_4__pow lpython_builtin [] __lpython_overloaded_4__pow Public), pow@__lpython_overloaded_5__pow: (ExternalSymbol 1 pow@__lpython_overloaded_5__pow 4 __lpython_overloaded_5__pow lpython_builtin [] __lpython_overloaded_5__pow Public), pow@__lpython_overloaded_6__pow: (ExternalSymbol 1 pow@__lpython_overloaded_6__pow 4 __lpython_overloaded_6__pow lpython_builtin [] __lpython_overloaded_6__pow Public), pow@__lpython_overloaded_7__pow: (ExternalSymbol 1 pow@__lpython_overloaded_7__pow 4 __lpython_overloaded_7__pow lpython_builtin [] __lpython_overloaded_7__pow Public), pow@__lpython_overloaded_8__pow: (ExternalSymbol 1 pow@__lpython_overloaded_8__pow 4 __lpython_overloaded_8__pow lpython_builtin [] __lpython_overloaded_8__pow Public), pow@__lpython_overloaded_9__pow: (ExternalSymbol 1 pow@__lpython_overloaded_9__pow 4 __lpython_overloaded_9__pow lpython_builtin [] __lpython_overloaded_9__pow Public), test_pow: (Subroutine (SymbolTable 2 {a: (Variable 2 a Local () () Default (Integer 4 []) Source Public Required .false.), a1: (Variable 2 a1 Local () (ConstantReal 4.500000 (Real 8 [])) Default (Real 8 []) Source Public Required .false.), a2: (Variable 2 a2 Local () (ConstantReal 2.300000 (Real 8 [])) Default (Real 8 []) Source Public Required .false.), abs: (ExternalSymbol 2 abs 4 abs lpython_builtin [] abs Private), b: (Variable 2 b Local () () Default (Integer 4 []) Source Public Required .false.), b1: (Variable 2 b1 Local () (ConstantLogical .true. (Logical 1 [])) Default (Logical 1 []) Source Public Required .false.), b2: (Variable 2 b2 Local () (ConstantLogical .false. (Logical 1 [])) Default (Logical 1 []) Source Public Required .false.), c1: (Variable 2 c1 Local () () Default (Complex 4 []) Source Public Required .false.), complex: (ExternalSymbol 2 complex 4 complex lpython_builtin [] complex Private), eps: (Variable 2 eps Local () () Default (Real 8 []) Source Public Required .false.), f1: (Variable 2 f1 Local () () Default (Real 4 []) Source Public Required .false.), f2: (Variable 2 f2 Local () () Default (Real 4 []) Source Public Required .false.), i1: (Variable 2 i1 Local () () Default (Integer 8 []) Source Public Required .false.), i2: (Variable 2 i2 Local () () Default (Integer 8 []) Source Public Required .false.), p: (Variable 2 p Local () () Default (Real 4 []) Source Public Required .false.), pow: (ExternalSymbol 2 pow 4 pow lpython_builtin [] pow Private), x: (Variable 2 x Local () (ConstantInteger 3 (Integer 4 [])) Default (Integer 4 []) Source Public Required .false.), y: (Variable 2 y Local () (ConstantReal 2.300000 (Real 8 [])) Default (Real 8 []) Source Public Required .false.)}) test_pow [] [(= (Var 2 eps) (ConstantReal 0.000000 (Real 8 [])) ()) (= (Var 2 a) (ConstantInteger 2 (Integer 4 [])) ()) (= (Var 2 b) (ConstantInteger 5 (Integer 4 [])) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((Var 2 a)) ((Var 2 b))] (Integer 4 []) () ()) Eq (ConstantInteger 32 (Integer 4 [])) (Logical 4 []) () ()) ()) (= (Var 2 a) (ConstantInteger 6 (Integer 4 [])) ()) (= (Var 2 b) (ConstantInteger 3 (Integer 4 [])) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((Var 2 a)) ((Var 2 b))] (Integer 4 []) () ()) Eq (ConstantInteger 216 (Integer 4 [])) (Logical 4 []) () ()) ()) (= (Var 2 a) (ConstantInteger 2 (Integer 4 [])) ()) (= (Var 2 b) (ConstantInteger 0 (Integer 4 [])) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((Var 2 a)) ((Var 2 b))] (Integer 4 []) () ()) Eq (ConstantInteger 1 (Integer 4 [])) (Logical 4 []) () ()) ()) (= (Var 2 a) (ConstantInteger 2 (Integer 4 [])) ()) (= (Var 2 b) (UnaryOp USub (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) (ConstantInteger -1 (Integer 4 []))) ()) (= (Var 2 a) (ConstantInteger 6 (Integer 4 [])) ()) (= (Var 2 b) (UnaryOp USub (ConstantInteger 4 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4 (Integer 4 []))) ()) (= (Var 2 i1) (Cast (ConstantInteger 2 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) ()) (= (Var 2 i2) (Cast (ConstantInteger 5 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_1__pow 2 pow [((Var 2 i1)) ((Var 2 i2))] (Integer 8 []) () ()) Eq (Cast (ConstantInteger 32 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) (Logical 4 []) () ()) ()) (= (Var 2 i1) (Cast (ConstantInteger 6 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) ()) (= (Var 2 i2) (Cast (UnaryOp USub (ConstantInteger 3 (Integer 4 [])) (Integer 4 []) (ConstantInteger -3 (Integer 4 []))) IntegerToInteger (Integer 8 []) ()) ()) (= (Var 2 f1) (Cast (BinOp (Cast (ConstantInteger 525346 (Integer 4 [])) IntegerToReal (Real 8 []) ()) Div (Cast (ConstantInteger 66456 (Integer 4 [])) IntegerToReal (Real 8 []) ()) (Real 8 []) () ()) RealToReal (Real 4 []) ()) ()) (= (Var 2 f2) (Cast (ConstantReal 3.000000 (Real 8 [])) RealToReal (Real 4 []) ()) ()) (= (Var 2 p) (FunctionCall 1 pow@__lpython_overloaded_2__pow 2 pow [((Var 2 f1)) ((Var 2 f2))] (Real 4 []) () ()) ()) (= (Var 2 f1) (FunctionCall 1 pow@__lpython_overloaded_4__pow 2 pow [((Var 2 a)) ((Var 2 f2))] (Real 4 []) () ()) ()) (= (Var 2 f1) (FunctionCall 1 pow@__lpython_overloaded_5__pow 2 pow [((Var 2 f2)) ((Var 2 a))] (Real 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_8__pow 2 pow [((Var 2 b1)) ((Var 2 b2))] (Integer 4 []) (ConstantInteger 1 (Integer 4 [])) ()) Eq (ConstantInteger 1 (Integer 4 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_8__pow 2 pow [((Var 2 b2)) ((Var 2 b1))] (Integer 4 []) (ConstantInteger 0 (Integer 4 [])) ()) Eq (ConstantInteger 0 (Integer 4 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_8__pow 2 pow [((Var 2 b1)) ((Var 2 b2))] (Integer 4 []) (ConstantInteger 1 (Integer 4 [])) ()) Eq (ConstantInteger 1 (Integer 4 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_8__pow 2 pow [((ConstantLogical .false. (Logical 1 []))) ((ConstantLogical .false. (Logical 1 [])))] (Integer 4 []) (ConstantInteger 1 (Integer 4 [])) ()) Eq (ConstantInteger 1 (Integer 4 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_3__pow 2 pow [((Var 2 a1)) ((Var 2 a2))] (Real 8 []) (ConstantReal 31.797193 (Real 8 [])) ()) Sub (ConstantReal 31.797193 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_3__pow 2 pow [((Var 2 a2)) ((Var 2 a1))] (Real 8 []) (ConstantReal 42.439989 (Real 8 [])) ()) Sub (ConstantReal 42.439989 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_6__pow 2 pow [((Var 2 x)) ((Var 2 y))] (Real 8 []) (ConstantReal 12.513503 (Real 8 [])) ()) Sub (ConstantReal 12.513503 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_7__pow 2 pow [((Var 2 y)) ((Var 2 x))] (Real 8 []) (ConstantReal 12.167000 (Real 8 [])) ()) Sub (ConstantReal 12.167000 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_6__pow 2 pow [((Var 2 x)) ((ConstantReal 5.500000 (Real 8 [])))] (Real 8 []) (ConstantReal 420.888346 (Real 8 [])) ()) Sub (ConstantReal 420.888346 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (Cast (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((ConstantInteger 2 (Integer 4 []))) ((UnaryOp USub (ConstantInteger 1 (Integer 4 [])) (Integer 4 []) (ConstantInteger -1 (Integer 4 []))))] (Integer 4 []) (ConstantReal 0.500000 (Real 8 [])) ()) IntegerToReal (Real 8 []) ()) Sub (ConstantReal 0.500000 (Real 8 [])) (Real 8 []) () ()))] (Real 8 []) () ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (Cast (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((ConstantInteger 6 (Integer 4 []))) ((UnaryOp USub (ConstantInteger 4 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4 (Integer 4 []))))] (Integer 4 []) (ConstantReal 0.000772 (Real 8 [])) ()) IntegerToReal (Real 8 []) ()) Sub (ConstantReal 0.000772 (Real 8 [])) (Real 8 []) () ()))] (Real 8 []) () ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (Cast (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((UnaryOp USub (ConstantInteger 3 (Integer 4 [])) (Integer 4 []) (ConstantInteger -3 (Integer 4 [])))) ((UnaryOp USub (ConstantInteger 5 (Integer 4 [])) (Integer 4 []) (ConstantInteger -5 (Integer 4 []))))] (Integer 4 []) (ConstantReal -0.004115 (Real 8 [])) ()) IntegerToReal (Real 8 []) ()) Add (ConstantReal 0.004115 (Real 8 [])) (Real 8 []) () ()))] (Real 8 []) () ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (Cast (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((ConstantInteger 6 (Integer 4 []))) ((UnaryOp USub (ConstantInteger 4 (Integer 4 [])) (Integer 4 []) (ConstantInteger -4 (Integer 4 []))))] (Integer 4 []) (ConstantReal 0.000772 (Real 8 [])) ()) IntegerToReal (Real 8 []) ()) Sub (ConstantReal 0.000772 (Real 8 [])) (Real 8 []) () ()))] (Real 8 []) () ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_3__pow 2 pow [((ConstantReal 4.500000 (Real 8 []))) ((ConstantReal 2.300000 (Real 8 [])))] (Real 8 []) (ConstantReal 31.797193 (Real 8 [])) ()) Sub (ConstantReal 31.797193 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_3__pow 2 pow [((ConstantReal 2.300000 (Real 8 []))) ((ConstantReal 0.000000 (Real 8 [])))] (Real 8 []) (ConstantReal 1.000000 (Real 8 [])) ()) Sub (ConstantReal 1.000000 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_3__pow 2 pow [((ConstantReal 2.300000 (Real 8 []))) ((UnaryOp USub (ConstantReal 1.500000 (Real 8 [])) (Real 8 []) (ConstantReal -1.500000 (Real 8 []))))] (Real 8 []) (ConstantReal 0.286687 (Real 8 [])) ()) Sub (ConstantReal 0.286687 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_6__pow 2 pow [((ConstantInteger 2 (Integer 4 []))) ((ConstantReal 3.400000 (Real 8 [])))] (Real 8 []) (ConstantReal 10.556063 (Real 8 [])) ()) Sub (ConstantReal 10.556063 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_6__pow 2 pow [((ConstantInteger 2 (Integer 4 []))) ((UnaryOp USub (ConstantReal 3.400000 (Real 8 [])) (Real 8 []) (ConstantReal -3.400000 (Real 8 []))))] (Real 8 []) (ConstantReal 0.094732 (Real 8 [])) ()) Sub (ConstantReal 0.094732 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_7__pow 2 pow [((ConstantReal 3.400000 (Real 8 []))) ((ConstantInteger 9 (Integer 4 [])))] (Real 8 []) (ConstantReal 60716.992766 (Real 8 [])) ()) Sub (ConstantReal 60716.992766 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_7__pow 2 pow [((ConstantReal 0.000000 (Real 8 []))) ((ConstantInteger 53 (Integer 4 [])))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Sub (ConstantReal 0.000000 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (Assert (Compare (FunctionCall 1 pow@__lpython_overloaded_0__pow 2 pow [((ConstantInteger 4 (Integer 4 []))) ((ConstantInteger 2 (Integer 4 [])))] (Integer 4 []) (ConstantInteger 16 (Integer 4 [])) ()) Eq (ConstantInteger 16 (Integer 4 [])) (Logical 4 []) (ConstantLogical .true. (Logical 4 [])) ()) ()) (Assert (Compare (FunctionCall 1 abs@__lpython_overloaded_0__abs 2 abs [((BinOp (FunctionCall 1 pow@__lpython_overloaded_7__pow 2 pow [((UnaryOp USub (ConstantReal 4235.000000 (Real 8 [])) (Real 8 []) (ConstantReal -4235.000000 (Real 8 [])))) ((ConstantInteger 52 (Integer 4 [])))] (Real 8 []) (ConstantReal 394800380598526378720936476336799774273305305904443955996320177992404102454228192853661558132283280490733920647962082901303487965679010854404517306573035287122910924343151116372519789002752.000000 (Real 8 [])) ()) Sub (ConstantReal 394800380598526378720936476336799774273305305904443955996320177992404102454228192853661558132283280490733920647962082901303487965679010854404517306573035287122910924343151116372519789002752.000000 (Real 8 [])) (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()))] (Real 8 []) (ConstantReal 0.000000 (Real 8 [])) ()) Lt (Var 2 eps) (Logical 4 []) () ()) ()) (= (Var 2 c1) (Cast (FunctionCall 1 complex@__lpython_overloaded_9__complex 2 complex [((ConstantInteger 4 (Integer 4 []))) ((ConstantInteger 5 (Integer 4 [])))] (Complex 8 []) (ConstantComplex 4.000000 5.000000 (Complex 8 [])) ()) ComplexToComplex (Complex 4 []) ()) ()) (= (Var 2 c1) (FunctionCall 1 pow@__lpython_overloaded_9__pow 2 pow [((Var 2 c1)) ((ConstantInteger 4 (Integer 4 [])))] (Complex 4 []) () ()) ())] Source Public Implementation () .false. .false.)}) [])
